### ISSUE TITLE
Is there a reason for RSNEAPOL::key(const key_type& value) to reset RSNEAPOL::key_t to 0?

### ISSUE THREAD

**Issue by asjadsyed** [mfontanini/libtins#133]
As I was building a Tins::RSNEAPOL packet, I noticed that the key_t for the packet I was building kept getting set back to Group Key (0) from Pairwise Key (1) which I set.
At first I thought I was making some sort of bounds-checking mistake and overwriting something, but I was able to track down the point where it changed by checking key_t's value at different points. 
The value of key_t is reset to 0 whenever the key for the packet is set. 

The line that resets it is line 232 in src/eapols.cpp
Here are lines 230-233 of src/eapol.cpp:
```
void RSNEAPOL::key(const key_type& value) {
    key_ = value;
    header_.key_t = 0; // right here
}
```

If there is any reasoning behind resetting it (related to the protocol or other), don't bother changing it. I just wanted to ask about it because it seemed like unintended behavior that might throw other programmers off during development. 
Thanks

**Comment by mfontanini** [comment]
Maybe when that was written, I was under the impression that key_t has to be zero when the key is present, but that seems bogus. That shouldn't be that way, right?

**Comment by asjadsyed** [comment]
From what I've gathered from Wireshark, key_t is an attribute in an RSNEAPOL packet that describes what type of key it is. A 0 in that field means that the packet is a Group Key and a 1 means that it's a Pairwise Key. 

You can create two packets that show the difference with this code and then look at the description Wireshark gives of the fields. The attribute is a 1-bit field in Wireshark under 802.1X Authentication > Key Information > Key Type.
```
#include <iostream>
#include <tins/tins.h>

using namespace Tins;
int main(int argc, char* argv[]) {
	PacketWriter writer("out.pcap", DataLinkType<RadioTap>());
	RadioTap pkt = RadioTap() / Dot11Data() / SNAP() / RSNEAPOL();

	pkt.rfind_pdu<RSNEAPOL>().key_t(0);
	writer.write(pkt);
	pkt.rfind_pdu<RSNEAPOL>().key_t(1);
	writer.write(pkt);
}
```
(If it seems odd that Wireshark is calling them a "Key (Group Message 2 of 2)" and just a "Key", it's because it doesn't have the other packets in the sequence and I haven't set any other fields that would otherwise help it deduce. )

But yeah, I don't think that field is there to say if a key is present. In every handshake capture I have, that field is always set.

**Comment by asjadsyed** [comment]
I found the exact descriptions of the bits in the [802.11-2012 Standards document](
http://standards.ieee.org/about/get/802/802.11.html) on pages 1245-1247 (section 11.6.2 EAPOL-Key frames). This is what it says about the Key Type bit:

> Key Type (bit 3) specifies whether this EAPOL-Key frame is part of a 4-Way Handshake
deriving a PTK.
i) The value 0 (Group/SMK) indicates the message is not part of a PTK derivation.
ii) The value 1 (Pairwise) indicates the message is part of a PTK derivation.

So I think it's safe to remove that line now, because setting a key does not mean that the RSNEAPOL PDU is for a Group Key
I've issued a pull request into branch develop: #140
